### PR TITLE
[dep] Use kuberay v0.5.2

### LINKED ISSUE #781: [dep] Use a released version of Ray operator
**What would you like to be cleaned**:
A released version of `github.com/ray-project/kuberay/ray-operator` should be used instead of `v0.0.0-20230516192117-9bc5d854bd31` once https://github.com/ray-project/kuberay/commit/9bc5d854bd316c46a79920bbd7b2887fb0c6d0af is part of one, probably v0.6.0.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

[dep] Use kuberay v0.5.2

#### Which issue(s) this PR fixes:

Fixes #781

#### Special notes for your reviewer:
Because `github.com/ray-project/kuberay` is not providing sub-module tags `ray-operator/v0.5.2` the go.mod reference is `v0.0.0-20230613204710-aeed3cdcbdcc`.
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/onsi/gomega v1.27.8
 	github.com/open-policy-agent/cert-controller v0.7.1-0.20230527042005-3b09cd39622f
 	github.com/prometheus/client_golang v1.16.0
-	github.com/ray-project/kuberay/ray-operator v0.0.0-20230516192117-9bc5d854bd31
+	github.com/ray-project/kuberay/ray-operator v0.0.0-20230613204710-aeed3cdcbdcc
```
Comment (tenzen-y) [inline go.mod:14]: Uhm... Maybe dependabot can not create PRs for Ray since not tagging 🤔

**Review by tenzen-y — COMMENTED** [review]
/lgtm

File: go.mod (hunk 1 above)
Comment (alculquicondor) [reply go.mod:14]: Sad :(
Can we leave a comment in the gomod and open an issue in kuberay?

**Review by alculquicondor — COMMENTED** [review]
/approve

File: go.mod (hunk 1 above)
Comment (tenzen-y) [reply go.mod:14]: @trasc Can you create an issue in KubeRay?

**Pull request merged**